frontend/src/cljs/imas_seamap/state_of_knowledge/events.cljs

```diff
@@ -64,101 +65,147 @@
              (not= network (get-in db [:state-of-knowledge :boundaries :amp :active-network]))
              (->
               (assoc-in [:state-of-knowledge :boundaries :amp :active-park] nil)
-              (assoc-in [:state-of-knowledge :boundaries :amp :active-network] network)))]
+              (assoc-in [:state-of-knowledge :boundaries :amp :active-network] network)))
+        park (get-in db [:state-of-knowledge :boundaries :amp :active-park])]
     {:db db
-     :dispatch-n [[:sok/get-habitat-statistics]
-                  [:sok/get-bathymetry-statistics]
-                  [:sok/get-habitat-observations]]}))
+     :dispatch-n (vec
+                  (concat
+                   [[:sok/get-habitat-statistics]
+                    [:sok/get-bathymetry-statistics]
+                    [:sok/get-habitat-observations]]
+                   (when (and network park) [[:sok/open-pill nil]])))}))
 
 (defn update-active-park [{:keys [db]} [_ {:keys [network] :as park}]]
-  (let [networks (get-in db [:state-of-knowledge :boundaries :amp :networks])
-        network (first-where #(= (:name %) network) networks)
+  (let [old-network (get-in db [:state-of-knowledge :boundaries :amp :active-network])
+        networks (get-in db [:state-of-knowledge :boundaries :amp :networks])
+        network (if park
+                  (first-where #(= (:name %) network) networks)
+                  old-network)
         db (-> db
                (assoc-in [:state-of-knowledge :boundaries :amp :active-network] network)
                (assoc-in [:state-of-knowledge :boundaries :amp :active-park] park))]
     {:db db
-     :dispatch-n [[:sok/get-habitat-statistics]
-                  [:sok/get-bathymetry-statistics]
-                  [:sok/get-habitat-observations]]}))
+     :dispatch-n (vec
```

Comment:
I haven't tried, but it might be worth a quick look to see what :dispatch-n does with a nil entry, that might save you a bit of vec-concat

Reply:
Actually, this might be a good opportunity for `cond->`; in conjunction with `conj` (which saves you the `vec` call)